test/e2e/parallel/usersignup_test.go

```diff
@@ -0,0 +1,214 @@
+			email:             "paul@hotel.com",
+			compliantUsername: "paul",
+		},
+		// Create another UserSignup with the original username matching the transformed username of the existing signup
+		{
+			username:          "paul",
+			email:             "paulathotel@hotel.com",
+			compliantUsername: "paul-2",
+		},
+		// Create a usersignup where the length of the username is greater than 20 chars, and is transformed by truncating
+		{
+			username:          "username-greater-than-20",
+			email:             "paulathotel@hotel.com",
+			compliantUsername: "username-greater-tha",
+		},
+		// Create a usersignup where the name is greater than 20 chars, but when truncating the username it has a forbidden suffix. Check the compliant username is replacing the suffix, instead of adding
+		{
+			username:          "username-with-admin-more-than-20-chars",
+			email:             "paulathotel@hotel.com",
+			compliantUsername: "username-with-ad-crt",
+		},
+	}
+
+	for _, testCase := range testCases {
+		newSignupRequestWithManualApprove(t, awaitilities, testCase)
+	}
+
+	testCasesPrefix := []TestCase{
+		// prefix with hyphen
+		{
+			username:          "-paul",
+			email:             "paul@hotel.com",
+			compliantUsername: "crt-%s-paul",
+		},
+		// prefix without delimiter
+		{
+			username:          "paul",
+			email:             "paul@hotel.com",
+			compliantUsername: "crt-%spaul",
+		},
+		// prefix as a name
+		{
+			username:          "",
+			email:             "paul@hotel.com",
+			compliantUsername: "crt-%s",
+		},
+	}
+
+	// Create another UserSignups with a forbidden prefix
+	for _, prefix := range []string{"kube", "openshift", "default", "redhat", "sandbox"} {
+		// prefix with hyphen
+		for _, testCase := range testCasesPrefix {
+			newSignupRequestWithManualApprove(t, awaitilities, TestCase{
+				username:          prefix + testCase.username,
+				email:             testCase.email,
+				compliantUsername: fmt.Sprintf(testCase.compliantUsername, prefix),
+			})
+		}
+	}
+
+	testCasesSuffix := []TestCase{
+		// suffix with hyphen
+		{
+			username:          "paul-",
+			email:             "paul@hotel.com",
+			compliantUsername: "paul-%s-crt",
+		},
+		// suffix without delimiter
+		{
+			username:          "paul",
+			email:             "paul@hotel.com",
+			compliantUsername: "paul%s-crt",
+		},
+		// suffix as a name
+		{
+			username:          "",
+			email:             "paul@hotel.com",
+			compliantUsername: "%s-crt",
+		},
+	}
+
+	// Create another UserSignups with a forbidden suffix
+	for _, suffix := range []string{"admin"} {
+		for _, testCase := range testCasesSuffix {
+			newSignupRequestWithManualApprove(t, awaitilities, TestCase{
+				username:          testCase.username + suffix,
+				email:             testCase.email,
+				compliantUsername: fmt.Sprintf(testCase.compliantUsername, suffix),
+			})
+		}
+	}
+}
+
+func newSignupRequestWithManualApprove(t *testing.T, awaitilities wait.Awaitilities, testCase TestCase) {
```

Comment:
Or maybe something like this would be even more accurate:
```suggestion
func assertComplaintUsernameForNewSignup(t *testing.T, awaitilities wait.Awaitilities, testCase TestCase) {
```